### ISSUE TITLE
[4.1] Cannot modify a user with backpack 4.1

### ISSUE THREAD

**Issue by DanielKimmich** [Laravel-Backpack/PermissionManager#225]
# Bug report
Symfony\Component\Debug\Exception\FatalErrorException
Can't use method return value in write context
http://crudcontact.telecom.int/admin/user

### What I did:
replace with new notation

public function update()
    {
        $this->crud->setRequest() = $this->crud->validateRequest();

### Backpack, Laravel, PHP, DB version:
php artisan backpack:version
### PHP VERSION:
PHP 7.2.11 (cli) (built: Oct 10 2018 02:04:07) ( ZTS MSVC15 (Visual C++ 2017) x64 )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies

### LARAVEL VERSION:
v6.18.2@9425a2f410d05d5bba493f62cff03854a8b19559

### BACKPACK VERSION:
4.1.x-dev@3eee77d99d429e257c4168071e70d8e1f5b92fec

**Comment by tabacitu** [comment]
I'm sorry @DanielKimmich but I can't figure out what you are trying to say here. Can you please rephrase? Please include:
- what page you were on
- what you were trying to do
- if you've modified things (and what)
- a screenshot of the error if possible

Cheers!

**Comment by pxpm** [comment]
I think here is the culprit, not tested, just looked around:

https://github.com/Laravel-Backpack/PermissionManager/blob/dab2c70cf1b1e43ab91fc75bcdb88eed7a63eddc/src/app/Http/Controllers/UserCrudController.php#L119

**Comment by tabacitu** [comment]
OOOOH! With 4.1 - didn't see that part. Now I get it 😄

**Comment by DanielKimmich** [comment]
Yes, that is the function that fails, function update() and store() in PermissionManager/src/app/Http/Controllers/UserCrudController.php. 

If I use the code like this, it gives me the following error
Symfony\Component\Debug\Exception\FatalThrowableError
Cannot access protected property Backpack\CRUD\app\Library\CrudPanel\CrudPanel::$request
http://crudcontact.telecom.int/admin/user/1

when I replace the "$this->crud->request" with a new notation, $this->crud->setRequest() = $this->crud->validateRequest();

The page does not open, Can't use method return value in write context

**Comment by tabacitu** [comment]
Indeed, I think the correct way to do it would be:
```php
$this->crud->setRequest($this->crud->validateRequest);
```

**Comment by DanielKimmich** [comment]
thank you very much it works.
I replace the following lines
       $this->crud->request = $this->crud->validateRequest();
       $this->crud->request = $this->handlePasswordInput($this->crud->request);
by
      $this->crud->setRequest($this->crud->validateRequest());
      $this->crud->setRequest($this->handlePasswordInput($this->crud->getRequest()));

regards.

**Comment by tabacitu** [comment]
Glad to hear that. We'll make sure to change this when we add 4.1 compatibility to the package. Which should be pretty soon 😄

**Comment by tabacitu** [comment]
Fixed in #228 . Thank you for the bug report @DanielKimmich & @mikiman2028 . To use this branch up until 4.1 is launched, you can change your ```composer.json``` to have:
```
"backpack/permissionmanager": "4.1.x-dev as 5.0.99"
```
and run ```composer update``` - that should pull that branch for you.

Closing the issue, since we have a working solution for it. Let's move the conversation to #228 if anything arises, please.

Cheers!
